packages/block-library/src/template-part/edit/name-panel.js

```diff
@@ -4,41 +4,22 @@
 import { useEntityProp } from '@wordpress/core-data';
 import { TextControl } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { cleanForSlug } from '@wordpress/url';
 
-export default function TemplatePartNamePanel( { postId, setAttributes } ) {
+export default function TemplatePartNamePanel( { postId } ) {
 	const [ title, setTitle ] = useEntityProp(
 		'postType',
 		'wp_template_part',
 		'title',
 		postId
 	);
-	const [ slug, setSlug ] = useEntityProp(
-		'postType',
-		'wp_template_part',
-		'slug',
-		postId
-	);
-	const [ status, setStatus ] = useEntityProp(
-		'postType',
-		'wp_template_part',
-		'status',
-		postId
-	);
 
 	return (
 		<div className="wp-block-template-part__name-panel">
 			<TextControl
-				label={ __( 'Name' ) }
-				value={ title || slug }
+				label={ __( 'Title' ) }
+				value={ title }
 				onChange={ ( value ) => {
 					setTitle( value );
-					const newSlug = cleanForSlug( value );
-					setSlug( newSlug );
```

Comment:
If we aren't updating the slug when we update the title, we run into places where slugs are very stale semantically.  That is, if I change the title of my "Header" to be "Not a header", the slug will still be `header`.  Similarly with custom creations the slugs will always be `template-part-#` regardless of what the title is.  Do you have any concerns about this?

Would there be any benefit to updating the slug corresponding to title?  So maybe "Custom Template Part" would have a slug of `custom-template-part-#` instead of `template-part-#`, and updating the title to "My Template Part" would update the slug to `my-template-part-#` ?

<hr />

Related to this change, I think we need to update the Nav Panel to show the Title instead of the Slug now.  Currently its just going to show up as the slug so the name isn't going to mean much to a user:

![Screen Shot 2021-01-07 at 2 34 01 PM](https://user-images.githubusercontent.com/28742426/103936529-86942f00-50f5-11eb-9a15-ee6c8d9525d2.png)

Reply:
> Do you have any concerns about this?

Good flag, and yes, I'm  aware of this, it's the same issue we have with reusable blocks. Ideally we need some kind of "modal" to choose the  title+slug before hitting the endpoint. So a UX change. (for both cases).

> Editing the slug as we edit the title

This doesn't make sense to me. I think the "slug" should be immutable because the template part can be already referenced in multiple places.

Reply:
That makes sense.  Thanks!

> This doesn't make sense to me. I think the "slug" should be immutable because the template part can be already referenced in multiple places.

Ahh right.  If we were referencing by `postId` it wouldn't matter, but since we are always referencing by `theme|slug` that would mess things up.  Good point!

Reply:
While I am definitely on board with getting rid of the auto-drafts, I am slightly concerned with removing the `postId` attribute entirely.  Isn't it the standard way to reference a CPT?  Keeping that attribute on the template part block allows us to:
* rename slugs semantically along with the titles
* not require them to be unique (allowing us to have easy access to both customized and uncustomized versions of the same template part)
* have a simple way to determine whether or not the template part block references an uncustomized vs. customized template part before even looking for it.

With removal of `postId` we are losing all of the above?

Reply:
In my mental modal, templates and template part don't reference CPTs. They reference templates (slugs) and the CPT is  just an implementation detail of where the template is saved if it's edited, if it's not edited otherwise, it's the theme one.

In other terms:

If I have "template a" referencing "template part x"
and "template b" referencing "template part x"

and I edit "template part x" in "template a", that change should apply to all templates referencing "template part x" (so also the template part referenced in "template b") and not just the one in the current template while if we did set a "post id" when we edit a template part, that change will only impact the current template.

For these reason, I think using "slugs" and dropping "ids" is more semantically correct.

Reply:
> and I edit "template part x" in "template a", that change should apply to all templates referencing "template part x" (so also the template part referenced in "template b") and not just the one in the current template while if we did set a "post id" when we edit a template part, that change will only impact the current template.

Very good point, It does solve that problem which is great!  In that light it does seem like the correct path forward, thanks for explaining again. 😁

Reply:
Thanks for challenging my explorations, that can only lead to better solutions.